components/persistor/lib/knex/db.ts

```diff
@@ -1116,234 +1122,41 @@ module.exports = function (PersistObjectTemplate) {
-                    return (propType.type === Array && propType.of.isObjectTemplate) ||
-                        (prop.match(/Persistor$/) && typeof props[propName.replace(/Persistor$/, '')] === 'object');
-                }
-
-                function generatePropertyChanges(prop, obj) {
-                    //When the property type is not an object template, need to compare the values.
-                    //for date and object types, need to compare the stringified values.
-                    var oldKey = '_ct_org_' + prop;
-                    const replaceNullValuesWithUndefined = function (k, v) { return v === null ? undefined : v; };
-                    if (!props[prop].type.isObjectTemplate && (obj[oldKey] != obj[prop] || ((props[prop].type === Date || props[prop].type === Object) &&
-                        JSON.stringify(obj[oldKey], replaceNullValuesWithUndefined) !== JSON.stringify(obj[prop], replaceNullValuesWithUndefined))))  {
-                        addChanges(prop, obj[oldKey], obj[prop], prop);
-                        obj[oldKey] = obj[prop];
-                    }
-                    //For one to one relations, we need to check the ids associated to the parent record.
-                    else if (props[prop].type.isObjectTemplate && obj[prop + 'Persistor'] && obj['_ct_org_' + prop] !== obj[prop + 'Persistor'].id) {
-                        addChanges(prop, obj[oldKey], obj[prop + 'Persistor'].id, getColumnName(prop, obj));
-                        obj[oldKey] = obj[prop + 'Persistor'].id;
-                    }
-                }
-
-                function getColumnName(prop, obj) {
-                    var schema = obj.__template__.__schema__;
-                    if (!schema || !schema.parents || !schema.parents[prop] || !schema.parents[prop].id)
-                        throw  new Error(obj.__template__.__name__ + '.' + prop + ' is missing a parents schema entry');
-                    return schema.parents[prop].id;
-                }
-
-                function addChanges(prop, originalValue, newValue, columnName) {
-                    objChanges.properties.push({
-                        name: prop,
-                        originalValue: originalValue,
-                        newValue: newValue,
-                        columnName: columnName
-                    });
-                }
-            }
-        }.bind(this)).then(function () {
-            (logger || this.logger).debug({component: 'persistor', module: 'api'}, 'end - transaction completed');
+        const knex = _.findWhere(this._db, {type: PersistObjectTemplate.DB_Knex}).connection;
+        let innerError;
+
+        // Apparently it seems like Knex takes an async callback, and implicitly waits for it to trigger knex.commit.
+        // This is confusing behavior
+        try {
+            innerError = await knex.transaction((knexTxn) => Transaction.transaction(this, notifyChanges, txn, knexTxn, logger)); // @TODO: Is Knex.transaction async?
```

Comment:
This TODO seems important, are there issues with awaiting a non-promise potentially here?

Reply:
erasing this comment - knex.transaction returns a promise as marked by the the typing definition for knex.transaction, so the answer is yes and don't think it needs any specific callout given we're awaiting the functionality in our code, and their definition is easily accessed

Reply:
side note - after googling a little it doesn't seem like there's any real measurable impact to awaiting a synchronous block of code, other than that it gets wrapped in a promise and then just resolves - which to me means some other promise that's hanging in wait might get a little cpu time, but it will not change the end result of this call.